src/test/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/TGContainerTest.groovy

```diff
@@ -177,4 +177,137 @@ class TGContainerTest extends ContainerTest {
+                        sleep(2)
+                                                    
+                        def response = httpRequest (
+                                            ignoreSslErrors: true,
+                                            acceptType: 'APPLICATION_JSON',
+                                            contentType: 'APPLICATION_JSON',
+                                            httpMode: 'POST',
+                                            url: "http://\${TG_HOSTNAME}:\${TG_API_PORT}/api/report/testCaseExecutions/filter?projectId=1&offset=0&limit=10",
+                                            customHeaders: [
+                                                [name: 'TestGuide-AuthKey', value: "${TG_AUTH_KEY}"]
+                                            ],
+                                            requestBody: '''
+                                            {
+                                                "testCaseName": ["test"],
+                                                "atxIds": ["1"],
+                                                "attributes": [
+                                                    {"key": "CustomAttribute", "values": ["Custom"]}
+                                                ],
+                                                "constants": [
+                                                    {"key": "CustomConstant", "values": ["Custom"]}
+                                                ]
+                                            }
+                                            '''
+                                       )
+                        if (response.status == 200) {
+                            println "Successfully retrieved the report from test.guide"
+                            println response.content
+                        } else {
+                            println "Retrieving the report from test.guide failed"
+                        }
+                        
+                    }
+                }
+            """.stripIndent()
+            WorkflowJob job = jenkins.createProject(WorkflowJob.class, "pipeline")
+            job.setDefinition(new CpsFlowDefinition(script, true))
+
+        when:
+            WorkflowRun run = jenkins.buildAndAssertStatus(Result.SUCCESS, job)
+
+        then:
+            jenkins.assertLogContains("Uploaded successfully", run)
+            jenkins.assertLogContains("Successfully retrieved the report from test.guide", run)
+            jenkins.assertLogContains('{"id":1,"projectId":1,"reportId":1,"testSuiteName":"SinglePackageExecution","testCaseName":"test"', run)
+            jenkins.assertLogContains('{"key":"BuildNumber","values":["42"]},{"key":"CustomConstant","values":["Custom"]}', run)
+            jenkins.assertLogContains('{"key":"CustomAttribute","values":["Custom"]}', run)
+    }
+
+    def "ttUploadReports: Test synchronous upload using additionalSettings and verify with test.guide request"() {
```

Comment:
looks mostly the same as the test above, better combine it to a parameterized test